logger/middleware.go

```diff
@@ -0,0 +1,68 @@
+package logger
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+)
+
+// name is the name of the application as recorded in latency metrics
+const name = "web"
+
+func getFirst(h http.Header, names ...string) string {
+	for _, name := range names {
+		if v := h.Get(name); v != "" {
+			return v
+		}
+	}
+	return ""
+}
+
+func Middleware(logger *logrus.Entry) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			logged := &loggedResponseWriter{
+				ResponseWriter: w,
+				Status:         200,
+			}
+			defer func(begin time.Time) {
+				// Try to get the real IP
+				remoteAddr := r.RemoteAddr
+				if realIP := getFirst(r.Header, "X-Real-IP", "X-Forwarded-For"); realIP != "" {
+					remoteAddr = realIP
```

Comment:
Would be cool to add this stuff to the request context

Reply:
Agree. It's not clear to me where it should live though since you need method/s to extract the values back out again. It means places that only receive context but want the values would have to weirdly depend on this package. What go-kit has seems pretty legit
https://github.com/go-kit/kit/blob/953e747656a7bbb5e1f998608b460458958b70cc/transport/http/request_response_funcs.go#L49